packages/web/src/layouts/governance/components/proposals-list/variable-select-box/VariableSelectBox.tsx

```diff
@@ -52,17 +58,36 @@ const VariableSelectBox: React.FC<VariableSelectBoxProps> = ({
     if (disabled === true) {
       return;
     }
-    setOpened(prev => !prev);
-  }, [disabled]);
+    onToggle(dropdownId, !isOpen);
+  }, [disabled, onToggle, dropdownId, isOpen]);
 
   const selectItem = useCallback(
     (value: string) => {
-      setOpened(false);
+      onToggle(dropdownId, false);
       onChange(value);
     },
-    [onChange],
+    [onChange, onToggle, dropdownId],
   );
 
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        isOpen &&
+        selectRef.current &&
+        dropdownRef.current &&
+        !selectRef.current.contains(event.target as Node) &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        onToggle(dropdownId, false);
+      }
+    };
```

Comment:
this condition is too complex. please consider breaking it down into separate functions